src/core_modules/capture-core/components/Pages/New/RegistrationDataEntry/RegistrationDataEntry.epics.js

```diff
@@ -92,13 +92,15 @@ const deriveEvents = ({ stages, enrollmentDate, incidentDate, programId, orgUnit
 export const startSavingNewTrackedEntityInstanceEpic: Epic = (action$: InputObservable, store: ReduxStore) =>
     action$.pipe(
         ofType(registrationFormActionTypes.NEW_TRACKED_ENTITY_INSTANCE_SAVE_START),
-        map(() => {
+        map((action) => {
             const { currentSelections: { orgUnitId, trackedEntityTypeId }, formsValues } = store.value;
             const values = formsValues['newPageDataEntryId-newTei'];
+            const formFoundation = action.payload?.formFoundation;
+            const formServerValues = formFoundation?.convertValues(values, convertToServer);
```

Comment:
This is a good start. We need to convert to client first though and then we can convert to server format.